frontend/src/scenes/events/eventsTableLogic.ts

```diff
@@ -299,24 +294,28 @@ export const eventsTableLogic = kea<eventsTableLogicType<ApiError, EventsTableLo
                     actions.setDelayedLoading()
                 }
             },
-            async ({ nextParams }, breakpoint) => {
+            async ({ queryLimits }, breakpoint: BreakPointFunction) => {
                 clearTimeout(values.pollTimeout)
 
                 const properties = [...values.properties, ...(props.fixedFilters?.properties || [])]
 
                 const params = {
                     ...(props.fixedFilters || {}),
                     properties,
-                    ...(nextParams || {}),
                     ...(values.eventFilter ? { event: values.eventFilter } : {}),
                     orderBy: [values.orderBy],
-                    after: values.afterParam,
+                    ...(queryLimits || {}),
+                    // we can send multiple after params, we should always filter for after props.fetchMonths ago
+                    // _and_ when we have events for more recent events
+                    // the N months ago is shown to the user
+                    after: [values.events[0]?.timestamp, now().subtract(values.months, 'months').toISOString()],
```

Comment:
I feel like this could be done simpler... Do we need the timestamp here at all? It's after all the first query here, not the poll query...